src/code/z_actor.c

```diff
@@ -457,439 +1244,3461 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+                flag = 2;
+            } else {
+                flag = (dmgFlags & 0x2000) ? 0x20 : 0;
+            }
+        }
+
+        actor->dropFlag |= flag;
+    }
+}
+
+void func_800BE33C(Vec3f* arg0, Vec3f* arg1, Vec3s* arg2, s32 arg3) {
+    f32 xDiff = arg1->x - arg0->x;
+    f32 zDiff = arg1->z - arg0->z;
+    f32 yDiff = arg3 ? (arg1->y - arg0->y) : (arg0->y - arg1->y);
+
+    arg2->y = Math_FAtan2F(zDiff, xDiff);
+    arg2->x = Math_FAtan2F(sqrtf(SQ(xDiff) + SQ(zDiff)), yDiff);
+}
+
+void func_800BE3D0(Actor* actor, s16 angle, Vec3s* arg2) {
+    f32 sp44;
+    f32 sp40;
+    f32 sp3C;
+    f32 sp38;
+    f32 sp34;
+    f32 sp30;
+    f32 sp2C;
+    s32 pad[3];
+
+    if (actor->floorPoly != NULL) {
+        CollisionPoly* floorPoly = actor->floorPoly;
+
+        sp44 = COLPOLY_GET_NORMAL(floorPoly->normal.x);
+        sp40 = COLPOLY_GET_NORMAL(floorPoly->normal.y);
+        sp3C = COLPOLY_GET_NORMAL(floorPoly->normal.z);
+
+        sp38 = Math_SinS(angle);
+        sp34 = Math_CosS(angle);
+        arg2->x = (s16)-Math_Atan2S((-(sp44 * sp38) - (sp3C * sp34)) * sp40, 1.0f);
+
+        sp2C = Math_SinS(angle - 0x3FF7);
+        sp30 = Math_CosS(angle - 0x3FF7);
+        arg2->z = (s16)-Math_Atan2S((-(sp44 * sp2C) - (sp3C * sp30)) * sp40, 1.0f);
+    }
+}
+
+void func_800BE504(Actor* actor, ColliderCylinder* collider) {
+    // Checks if was hit by either DMG_NORMAL_ARROW, DMG_FIRE_ARROW, DMG_ICE_ARROW, DMG_LIGHT_ARROW or DMG_DEKU_BUBBLE
+    if ((collider->info.acHitInfo->toucher.dmgFlags & (0x10000 | 0x2000 | 0x1000 | 0x800 | 0x20))) {
+        actor->world.rot.y = collider->base.ac->shape.rot.y;
+    } else {
+        actor->world.rot.y = Actor_YawBetweenActors(collider->base.ac, actor);
+    }
+}
+
+void func_800BE568(Actor* actor, ColliderSphere* collider) {
+    if (collider->info.acHitInfo->toucher.dmgFlags & (0x10000 | 0x2000 | 0x1000 | 0x800 | 0x20)) {
+        actor->world.rot.y = collider->base.ac->shape.rot.y;
+    } else {
+        actor->world.rot.y = Actor_YawBetweenActors(collider->base.ac, actor);
+    }
+}
+
+void func_800BE5CC(Actor* actor, ColliderJntSph* collider, s32 colliderIndex) {
+    if (collider->elements[colliderIndex].info.acHitInfo->toucher.dmgFlags &
+        (0x10000 | 0x2000 | 0x1000 | 0x800 | 0x20)) {
+        actor->world.rot.y = collider->base.ac->shape.rot.y;
+    } else {
+        actor->world.rot.y = Actor_YawBetweenActors(collider->base.ac, actor);
+    }
+}
+
+s32 func_800BE63C(struct EnBox* chest) {
+    if ((chest->unk_1F1 == 5) || (chest->unk_1F1 == 6) || (chest->unk_1F1 == 7) || (chest->unk_1F1 == 8) ||
+        (chest->unk_1F1 == 0xC)) {
+        return true;
+    }
+    return false;
+}
+
+TexturePtr D_801AEFA8[] = {
+    gameplay_keep_Tex_091DE0,
+    gameplay_keep_Tex_091FE0,
+    gameplay_keep_Tex_0921E0,
+    gameplay_keep_Tex_0923E0,
+};
+
+void func_800BE680(GlobalContext* globalCtx, Actor* actor, Vec3f limbPos[], s16 arg3, f32 arg4, f32 arg5, f32 arg6,
```

Comment:
This is used in a lot of places, looks like it's for drawing effects on enemies mostly?

Reply:
This is the first function I'm docing once this PR is in. Yeah it's for drawing various generic effects on enemies, like when they're shot by light arrows and start glowing, or they get frozen, or they are thawed and start steaming, or when they get hit by zora's electric stun and are in shock. Basically, this enum needs to be put into z64actor.h and made global, and extended to account for all cases:
https://github.com/zeldaret/mm/blob/44ccdbe0d5bc2e3aad872613070106dae1de7a4e/src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.h#L23-L28